wagtail/images/blocks.py

```diff
@@ -51,3 +56,133 @@ def htmldiff(self):
                 "image_b": self.val_b,
             },
         )
+
+
+class ImageBlock(StructBlock):
+    image = ImageChooserBlock(required=True)
+    decorative = BooleanBlock(
+        default=False, required=False, label="Image is decorative"
+    )
+    alt_text = CharBlock(required=False)
+
+    def get_searchable_content(self, value):
+        return []
+
+    def _struct_value_to_image(self, struct_value):
+        image = struct_value.get("image")
+        decorative = struct_value.get("decorative")
+        if image:
+            # If the image is decorative, set alt_text to an empty string
+            image.contextual_alt_text = (
+                "" if decorative else struct_value.get("alt_text", "")
+            )
+            image.decorative = decorative
+        return image
+
+    def to_python(self, value):
+        # For backward compatibility with ImageChooserBlock
+        if isinstance(value, int):
+            from wagtail.images import get_image_model
+
+            Image = get_image_model()
+            image = Image.objects.get(
+                id=value
+            )  # To retrieve specific image instance from db
+            struct_value = {"image": image, "decorative": False, "alt_text": ""}
+        else:
+            struct_value = super().to_python(value)
+        return self._struct_value_to_image(struct_value)
+
+    def bulk_to_python(self, values):
+        # For backward compatibility with ImageChooserBlock
+        struct_values = []
+        for value in values:
+            if isinstance(value, int):
+                from wagtail.images import get_image_model
+
+                Image = get_image_model()
+                image = Image.objects.get(
+                    id=value
+                )  # To retrieve specific image instance from db
+                struct_value = {"image": image, "decorative": False, "alt_text": ""}
+                struct_values.append(struct_value)
+            else:
+                struct_values = super().bulk_to_python(values)
+        return [
+            self._struct_value_to_image(struct_value) for struct_value in struct_values
+        ]
+
+    def value_from_datadict(self, data, files, prefix):
+        struct_value = super().value_from_datadict(data, files, prefix)
+        return self._struct_value_to_image(struct_value)
+
+    def clean(self, value):
+        if value is None:
+            raise StructBlockValidationError(
```

Comment:
I’m surprised this isn’t calling the `clean` method from StructBlock? `result = super().clean(value)`

Reply:
The problem with that is that using the `super().clean(value)` would fail. Reason being is that the superclass method calls that with the expectation that the value received will be a `dict`, and calls the `.items()` method on it. That will not be the case for us, where the value is supposed to be an Image instance.